sos/policies/runtimes/crio.py

```diff
@@ -0,0 +1,80 @@
+# Copyright (C) 2021 Red Hat, Inc., Nadia Pinaeva <npinaeva@redhat.com>
+
+# This file is part of the sos project: https://github.com/sosreport/sos
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# version 2 of the GNU General Public License.
+#
+# See the LICENSE file in the source distribution for further information.
+
+from sos.policies.runtimes import ContainerRuntime
+from sos.utilities import sos_get_command_output
+from pipes import quote
+
+
+class CrioContainerRuntime(ContainerRuntime):
+    """Runtime class to use for systems running crio"""
+
+    name = 'crio'
+    binary = 'crictl'
+
+    def get_containers(self, get_all=False):
+        """Get a list of containers present on the system.
+
+        :param get_all: If set, include stopped containers as well
+        :type get_all: ``bool``
+        """
+        containers = []
+        _cmd = "%s ps %s" % (self.binary, '-a' if get_all else '')
+        if self.active:
+            out = sos_get_command_output(_cmd, chroot=self.policy.sysroot)
+            if out['status'] == 0:
+                for ent in out['output'].splitlines()[1:]:
+                    ent = ent.split()
+                    # takes the form (container_id, container_name)
+                    containers.append((ent[0], ent[-3]))
+        return containers
+
+    def get_images(self):
+        """Get a list of images present on the system
+
+        :returns: A list of 2-tuples containing (image_name, image_id)
+        :rtype: ``list``
+        """
+        images = []
+        if self.active:
+            out = sos_get_command_output("%s images" % self.binary,
+                                         chroot=self.policy.sysroot)
+            if out['status'] == 0:
+                for ent in out['output'].splitlines():
+                    ent = ent.split()
+                    # takes the form (image_name, image_id)
+                    images.append((ent[0] + ':' + ent[1], ent[2]))
```

Comment:
Here we rely on `len(ent) > 2`. Is it a safe assumption in either use case? (I don't know `crio` enough, hence asking)

Reply:
`crio images` has the following columns in the output
`IMAGE                                                     TAG                 IMAGE ID            SIZE`
So it's the same as parsing output of any other command

Reply:
Is there a condition where `crio images` reports an error message, but still gives a `0` exit code? I don't believe we've had this with other runtime executables, but we have seen this in other collections in general.

Reply:
I definitely can't guarantee that, but I also have no other means to verify its output, even if I check for `ent` len, in case of error it'll probably have more than 3 words, so I'll just return the wrong `images` list